doc/source/whatsnew/v1.1.0.rst

```diff
@@ -328,6 +328,7 @@ Other enhancements
 - :meth:`DataFrame.to_html` and :meth:`DataFrame.to_string`'s ``col_space`` parameter now accepts a list or dict to change only some specific columns' width (:issue:`28917`).
 - :meth:`DataFrame.to_excel` can now also write OpenOffice spreadsheet (.ods) files (:issue:`27222`)
 - :meth:`~Series.explode` now accepts ``ignore_index`` to reset the index, similarly to :meth:`pd.concat` or :meth:`DataFrame.sort_values` (:issue:`34932`).
+- :meth:`BooleanArray._from_sequence_of_strings` now accepts "0", "0.0", "1", "1.0" (:issue:`34859`)
```

Comment:
This isn't a public method, so we shouldn't mention it in the release notes. Mention in in terms of something like `read_csv` now accepting strings like "0", "0.0", ... as convertable to the nullable boolean dtype.